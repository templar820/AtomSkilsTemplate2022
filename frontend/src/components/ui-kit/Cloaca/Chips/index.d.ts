import * as React from 'react';
import { EmotionStylesType } from 'lib-ui';

export interface ChipsProps {
    /** Defines color scheme of component */
    colorScheme?: 'gray' | 'white' | 'primary' | 'red';
    /** Defines color of component */
    color?: string;
    /** Defines text or any node that will be placed in a body of the component */
    children?: React.ReactNode;
    /** Callback that fires when component is closed, func(e) */
    onClose?: (event: React.MouseEvent) => void;
    /** Callback that fires on mouse click, func(e) */
    onClick?: (event: React.MouseEvent) => void;
    /** Defines whether component is disabled
     * isDisabled will be deprecated in future versions
     */
    disabled?: boolean;
    /** Defines whether component is checkable */
    isCheckable?: boolean;
    /** Defines whether component is checked */
    checked?: boolean;
    /** Callback that fires when checkbox's state has changed, func(e) */
    onCheckChange?: (event: React.MouseEvent) => void;
    /** Defines whether component is round */
    round?: boolean;
    /** Defines the size of border-radius */
    radius?: string;
    /** Defines component's font-size */
    fontSize?: string | number;
    /** Defines coponent's additional style */
    style?: EmotionStylesType;
    /** Defines props of close button */
    closeBtnProps?: object;
    /** Defines additional component css */
    css?: EmotionStylesType;
    /** Defines additional className */
    className?: string;
}

declare const Chips: React.ComponentType<ChipsProps>;

export default Chips;
