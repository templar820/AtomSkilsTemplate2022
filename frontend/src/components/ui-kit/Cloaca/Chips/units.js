import { css } from '@emotion/core';
import styled from '@emotion/styled';

import Icon from 'lib-ui/InlineIcons';

import { hexToRGBA, pickColor } from 'lib-root/utils/styleMixins';
import applyEmotionStyle from 'lib-root/utils/applyEmotionStyle';

const baseStyles = ({ colors, fontSize, onClose }) => css`
    display: inline-flex;
    justify-content: center;
    min-width: ${onClose ? '20px' : 'auto'};
    min-height: 40px;
    padding: 7px 12px;
    border: 1px solid;
    border-radius: 1px;
    font-size: ${typeof fontSize === 'number' ? fontSize + 'px' : fontSize};
    line-height: 1.6;
    cursor: pointer;
    transition: 0.3s all;
    background-color: ${hexToRGBA(colors.GrayScale_100, 0.5)};
    color: ${colors.GrayScale_0};
    white-space: nowrap;
    &.hover,
    &:hover {
        background-color: ${hexToRGBA(colors.GrayScale_0, 0.2)};
        color: ${colors.GrayScale_700};
    }
    &:focus {
        outline: 0;
    }
    &:disabled {
        background-color: ${colors.GrayScale_200};
        border-color: ${colors.GrayScale_200};
        cursor: not-allowed;
        color: ${colors.GrayScale_0};
    }
`;

const colorSchemeStyles = {
    gray: ({ colors, color: customColor }) => css`
        background-color: ${hexToRGBA(pickColor(colors.GrayScale_100, null, customColor), 0.5)};
        border-color: ${hexToRGBA(pickColor(colors.GrayScale_100, null, customColor), 0.5)};
        color: ${colors.GrayScale_700};
        &.hover,
        &:hover {
            background-color: ${hexToRGBA(pickColor(colors.GrayScale_100, null, customColor), 0.5)};
            border-color: ${pickColor(colors.GrayScale_200, null, customColor)};
            box-shadow: 0 0 4px ${hexToRGBA(colors.GrayScale_700, 0.25)};
            color: ${colors.GrayScale_700};
        }
    `,
    white: ({ colors, color: customColor }) => css`
        background-color: ${hexToRGBA(pickColor(colors.GrayScale_0, null, customColor), 0.5)};
        border-color: ${hexToRGBA(pickColor(colors.GrayScale_100, null, customColor), 0.5)};
        color: ${colors.GrayScale_700};
        &.hover,
        &:hover {
            background-color: ${hexToRGBA(pickColor(colors.GrayScale_0, null, customColor), 0.5)};
            border-color: ${pickColor(colors.GrayScale_200, null, customColor)};
            box-shadow: 0 0 5px ${hexToRGBA(colors.GrayScale_700, 0.25)};
            color: ${colors.GrayScale_700};
        }
    `,
    red: ({ colors, color: customColor }) => css`
        background-color: ${hexToRGBA(pickColor(colors.errorAccent, null, customColor), 0.2)};
        border-color: ${pickColor(colors.errorAccent, null, customColor)};
        color: ${pickColor(colors.errorAccent, null, customColor)};
        &.hover,
        &:hover {
            background-color: ${hexToRGBA(pickColor(colors.errorAccent, null, customColor), 0.1)};
            color: ${colors.GrayScale_700};
        }
        &:disabled {
            background-color: ${colors.GrayScale_100};
            color: ${hexToRGBA(colors.GrayScale_700, 0.5)};
        }
    `,
    primary: ({ colors, color: customColor }) => css`
        background-color: ${pickColor(colors.primary, null, customColor)};
        border-color: ${pickColor(colors.primary, null, customColor)};
        color: ${colors.GrayScale_0};
        &.hover,
        &:hover {
            background-color: ${hexToRGBA(pickColor(colors.primary, null, customColor), 0.2)};
            color: ${colors.GrayScale_700};
        }
    `,
    checked: ({ colors, color: customColor }) => css`
        background-color: ${pickColor(colors.GrayScale_0, null, customColor)};
        border-color: ${pickColor(colors.GrayScale_0, null, customColor)};
        color: ${colors.GrayScale_700};
        &.hover,
        &:hover {
            background-color: ${hexToRGBA(pickColor(colors.primary, null, customColor), 0.2)};
            color: ${colors.GrayScale_700};
        }
    `,
    selected: ({ colors, color: customColor }) => css`
        background-color: ${pickColor(colors.primary, null, customColor)};
        border-color: ${pickColor(colors.primary, null, customColor)};
        color: ${colors.GrayScale_0};
        &.hover,
        &:hover {
            background-color: ${hexToRGBA(pickColor(colors.primary, null, customColor), 0.2)};
            border-color: ${hexToRGBA(pickColor(colors.primary, null, customColor), 0.2)};
            color: ${colors.GrayScale_700};
        }
    `
};

const StyledChips = styled.div`
    ${baseStyles}
    ${({ dashed }) => dashed && 'border-style: dashed'};
    ${({ round, radius }) => round && `border-radius: ${radius}`};
    ${(props) => (colorSchemeStyles[props.colorScheme || 'gray'] || colorSchemeStyles.primary)(props)}
    ${({ isCheckable, checked }) =>
        isCheckable && (checked ? colorSchemeStyles['selected'] : colorSchemeStyles['checked'])}
    ${({ css, ...restProps }) => applyEmotionStyle(css, restProps)};
`;

const StyledCloseBtn = styled(Icon)`
    width: 20px;
    height: 20px;
    min-width: 20px;
    margin-left: 12px;
    &.hover,
    &:hover {
        transition: 0.3s all;
        transform: scale(1.1);
    }
`;

const StyledContentWrapper = styled.div`
    display: inline-flex;
    justify-content: center;
    align-items: center;
    min-width: 40px;
`;

export { StyledChips, StyledCloseBtn, StyledContentWrapper };
