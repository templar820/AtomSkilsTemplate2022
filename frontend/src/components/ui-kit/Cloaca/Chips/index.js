import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { ColorsContext } from 'lib-root/colors';

import { StyledChips, StyledContentWrapper, StyledCloseBtn } from './units';
import { setClassName } from 'lib-root/utils/styleMixins';
import { resolveProps } from 'lib-root/utils';

/**
 * `Chips` - это компактные элементы, представляющие входные данные, атрибут или действие.
 */
class Chips extends Component {
    changeCheck = (e) => {
        const { onCheckChange, onClick, isDisabled, disabled: propDisabled } = this.props;
        const disabled = resolveProps(propDisabled, isDisabled);

        if (disabled) return;
        if (onCheckChange) onCheckChange(e);
        if (onClick) onClick(e);
    };

    getCloseBtn() {
        const { onClose, closeBtnProps } = this.props;
        return onClose && <StyledCloseBtn onClick={onClose} icon={'Cross_exit'} {...closeBtnProps} />;
    }

    renderContent() {
        const { children } = this.props;
        return (
            <StyledContentWrapper>
                {children}
                {this.getCloseBtn()}
            </StyledContentWrapper>
        );
    }

    render() {
        const { onCheckChange, ...rest } = this.props;
        return (
            <ColorsContext.Consumer>
                {(colors) => (
                    <StyledChips
                        {...{ colors, ...rest }}
                        onClick={this.changeCheck}
                        className={setClassName({ props: this.props, name: 'chips' })}>
                        {this.renderContent()}
                    </StyledChips>
                )}
            </ColorsContext.Consumer>
        );
    }
}
Chips.displayName = 'Chips';
Chips.propTypes = {
    /** Defines color scheme of component */
    colorScheme: PropTypes.oneOf(['gray', 'white', 'primary', 'red']),
    /** Defines color of component */
    color: PropTypes.string,
    /** Defines text or any node that will be placed in a body of the component */
    children: PropTypes.node,
    /** Callback that fires when component is closed, func(e) */
    onClose: PropTypes.func,
    /** Callback that fires on mouse click, func(e) */
    onClick: PropTypes.func,
    /** Defines whether component is disabled
     * isDisabled will be deprecated in future versions
     */
    disabled: PropTypes.bool,
    /** Defines whether component is checkable */
    isCheckable: PropTypes.bool,
    /** Defines whether component is checked */
    checked: PropTypes.bool,
    /** Callback that fires when checkbox's state has changed, func(e) */
    onCheckChange: PropTypes.func,
    /** Defines whether component is round */
    round: PropTypes.bool,
    /** Defines the size of border-radius */
    radius: PropTypes.string,
    /** Defines component's font-size */
    fontSize: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    /** Defines coponent's additional style */
    style: PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.string]),
    /** Defines props of close button */
    closeBtnProps: PropTypes.object,
    /** Defines additional component css */
    css: PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.string]),
    /** Defines additional className */
    className: PropTypes.string
};

Chips.defaultProps = {
    colorScheme: 'primary',
    disabled: false,
    round: false,
    isCheckable: false,
    checked: false,
    radius: '25px',
    fontSize: '14px'
};

export default Chips;
