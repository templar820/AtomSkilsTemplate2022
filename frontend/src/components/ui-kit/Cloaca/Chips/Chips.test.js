import React from 'react';

import serializer from 'jest-emotion';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import Chips from './';

expect.addSnapshotSerializer(serializer);

Enzyme.configure({ adapter: new Adapter() });

const funcMock = jest.fn();

const props = {
    onClick: () => funcMock('onClick'),
    onClose: () => funcMock('onClose'),
    onCheckChange: () => funcMock('onCheckChange'),
    children: <div id="children">Chips children</div>,
    color: '#000',
    css: { margin: '18px' },
    style: { background: '#fff', padding: 10 },
    colorScheme: 'gray',
    radius: '15px',
    fontSize: '15px'
};

describe('<Chips />', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(<Chips {...props} />);
    });

    afterEach(() => {
        wrapper.unmount();
    });

    test('should match snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('should render correctly', () => {
        expect(wrapper.find('div[id="children"]')).toBeTruthy();
        expect(wrapper.find('Chips children')).toBeTruthy();

        expect(wrapper.props()).toHaveProperty('fontSize', props.fontSize);
        expect(wrapper.props()).toHaveProperty('radius', props.radius);
        expect(wrapper.props()).toHaveProperty('color', props.color);
        expect(wrapper.props()).toHaveProperty('colorScheme', props.colorScheme);
        expect(wrapper.props()).toHaveProperty('css', props.css);
    });

    test('should component disable', () => {
        wrapper.setProps({ disabled: true });
        wrapper.update();
        wrapper.simulate('click');
        expect(funcMock).not.toHaveBeenCalled();
    });

    test('should call onClick function', () => {
        wrapper.simulate('click');
        expect(funcMock).toHaveBeenCalledWith('onClick');
    });

    test('should call onChange function', () => {
        wrapper.simulate('change', { target: { value: true } });
        expect(funcMock).toHaveBeenCalledWith('onCheckChange');
    });

    test('should have onClose function', () => {
        wrapper.find('svg').simulate('click');
        expect(funcMock).toHaveBeenCalledWith('onClose');
    });
});
