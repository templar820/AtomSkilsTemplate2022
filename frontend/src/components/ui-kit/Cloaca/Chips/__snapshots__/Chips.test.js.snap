// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chips /> should match snapshot 1`] = `
.emotion-11 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-width: 20px;
  min-height: 40px;
  padding: 7px 12px;
  border: 1px solid;
  border-radius: 1px;
  font-size: 15px;
  line-height: 1.6;
  cursor: pointer;
  -webkit-transition: 0.3s all;
  transition: 0.3s all;
  background-color: rgba(237,237,237,0.5);
  color: #ffffff;
  white-space: nowrap;
  background-color: rgba(0,0,0,0.5);
  border-color: rgba(0,0,0,0.5);
  color: #333333;
  margin: 18px;
}

.emotion-11.hover,
.emotion-11:hover {
  background-color: rgba(255,255,255,0.2);
  color: #333333;
}

.emotion-11:focus {
  outline: 0;
}

.emotion-11:disabled {
  background-color: #D8D8D8;
  border-color: #D8D8D8;
  cursor: not-allowed;
  color: #ffffff;
}

.emotion-11.hover,
.emotion-11:hover {
  background-color: rgba(0,0,0,0.5);
  border-color: #000;
  box-shadow: 0 0 4px rgba(51,51,51,0.25);
  color: #333333;
}

.emotion-9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 40px;
}

.emotion-3 {
  width: 20px;
  height: 20px;
  min-width: 20px;
  margin-left: 12px;
}

.emotion-3.hover,
.emotion-3:hover {
  -webkit-transition: 0.3s all;
  transition: 0.3s all;
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

.emotion-1 {
  fill: currentColor;
  width: 32px;
  min-width: 32px;
  height: 32px;
  width: 20px;
  height: 20px;
  min-width: 20px;
  margin-left: 12px;
}

.emotion-1.hover,
.emotion-1:hover {
  -webkit-transition: 0.3s all;
  transition: 0.3s all;
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

<Chips
  checked={false}
  color="#000"
  colorScheme="gray"
  css={
    Object {
      "margin": "18px",
    }
  }
  disabled={false}
  fontSize="15px"
  isCheckable={false}
  onCheckChange={[Function]}
  onClick={[Function]}
  onClose={[Function]}
  radius="15px"
  round={false}
  style={
    Object {
      "background": "#fff",
      "padding": 10,
    }
  }
>
  <StyledChips
    checked={false}
    className="CLR-chips"
    color="#000"
    colorScheme="gray"
    colors={
      Object {
        "GrayScale_0": "#ffffff",
        "GrayScale_100": "#EDEDED",
        "GrayScale_150": "#E5E5E5",
        "GrayScale_200": "#D8D8D8",
        "GrayScale_50": "#fafafa",
        "GrayScale_500": "#999999",
        "GrayScale_55": "#f6f6f6",
        "GrayScale_600": "#7C7C7C",
        "GrayScale_700": "#333333",
        "GrayScale_900": "#000000",
        "error": "#F62434",
        "errorAccent": "#EB5A40",
        "info": "#34AAF2",
        "infoAccent": "#5BD9E5",
        "primary": "#00B956",
        "primaryAccent": "#10E272",
        "secondary": "#731982",
        "secondaryAccent": "#A500BF",
        "tertiary": "#444189",
        "tertiaryAccent": "#404D46",
        "warning": "#FFA717",
        "warningAccent": "#F36F20",
      }
    }
    css={
      Object {
        "margin": "18px",
      }
    }
    disabled={false}
    fontSize="15px"
    isCheckable={false}
    onClick={[Function]}
    onClose={[Function]}
    radius="15px"
    round={false}
    style={
      Object {
        "background": "#fff",
        "padding": 10,
      }
    }
  >
    <div
      checked={false}
      className="CLR-chips emotion-11 emotion-12"
      color="#000"
      disabled={false}
      fontSize="15px"
      onClick={[Function]}
      onClose={[Function]}
      radius="15px"
      style={
        Object {
          "background": "#fff",
          "padding": 10,
        }
      }
    >
      <StyledContentWrapper>
        <div
          className="emotion-9 emotion-10"
        >
          <div
            id="children"
          >
            Chips children
          </div>
          <StyledCloseBtn
            color="currentColor"
            disabled={false}
            h={null}
            height={null}
            icon="Cross_exit"
            onClick={[Function]}
            w={null}
            width={null}
          >
            <withColors(InlineIcons)
              className="emotion-3 emotion-0"
              color="currentColor"
              disabled={false}
              h={null}
              height={null}
              icon="Cross_exit"
              onClick={[Function]}
              w={null}
              width={null}
            >
              <InlineIcons
                className="emotion-3 emotion-0"
                color="currentColor"
                colors={
                  Object {
                    "GrayScale_0": "#ffffff",
                    "GrayScale_100": "#EDEDED",
                    "GrayScale_150": "#E5E5E5",
                    "GrayScale_200": "#D8D8D8",
                    "GrayScale_50": "#fafafa",
                    "GrayScale_500": "#999999",
                    "GrayScale_55": "#f6f6f6",
                    "GrayScale_600": "#7C7C7C",
                    "GrayScale_700": "#333333",
                    "GrayScale_900": "#000000",
                    "error": "#F62434",
                    "errorAccent": "#EB5A40",
                    "info": "#34AAF2",
                    "infoAccent": "#5BD9E5",
                    "primary": "#00B956",
                    "primaryAccent": "#10E272",
                    "secondary": "#731982",
                    "secondaryAccent": "#A500BF",
                    "tertiary": "#444189",
                    "tertiaryAccent": "#404D46",
                    "warning": "#FFA717",
                    "warningAccent": "#F36F20",
                  }
                }
                disabled={false}
                h={null}
                height={null}
                icon="Cross_exit"
                onClick={[Function]}
                w={null}
                width={null}
              >
                <Styled(svg)
                  className="emotion-3 emotion-0 CLR-inline-icons"
                  color="currentColor"
                  colors={
                    Object {
                      "GrayScale_0": "#ffffff",
                      "GrayScale_100": "#EDEDED",
                      "GrayScale_150": "#E5E5E5",
                      "GrayScale_200": "#D8D8D8",
                      "GrayScale_50": "#fafafa",
                      "GrayScale_500": "#999999",
                      "GrayScale_55": "#f6f6f6",
                      "GrayScale_600": "#7C7C7C",
                      "GrayScale_700": "#333333",
                      "GrayScale_900": "#000000",
                      "error": "#F62434",
                      "errorAccent": "#EB5A40",
                      "info": "#34AAF2",
                      "infoAccent": "#5BD9E5",
                      "primary": "#00B956",
                      "primaryAccent": "#10E272",
                      "secondary": "#731982",
                      "secondaryAccent": "#A500BF",
                      "tertiary": "#444189",
                      "tertiaryAccent": "#404D46",
                      "warning": "#FFA717",
                      "warningAccent": "#F36F20",
                    }
                  }
                  disabled={false}
                  height="32px"
                  onClick={[Function]}
                  width="32px"
                >
                  <svg
                    className="emotion-0 CLR-inline-icons emotion-1 emotion-2"
                    color="currentColor"
                    disabled={false}
                    height="32px"
                    onClick={[Function]}
                    width="32px"
                  >
                    <use
                      xlinkHref="#Cross_exit"
                    />
                  </svg>
                </Styled(svg)>
              </InlineIcons>
            </withColors(InlineIcons)>
          </StyledCloseBtn>
        </div>
      </StyledContentWrapper>
    </div>
  </StyledChips>
</Chips>
`;
